src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/CompositeMLDsa/CompositeMLDsaFactoryTests.cs

```diff
@@ -1,21 +1,157 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using Internal.Cryptography;
 using Xunit;
 
 namespace System.Security.Cryptography.Tests
 {
     public static class CompositeMLDsaFactoryTests
     {
+        // TODO test doesn't belong here, move to different class
+        [Theory]
+        [MemberData(nameof(CompositeMLDsaTestData.SupportedAlgorithmIetfVectorsTestData), MemberType = typeof(CompositeMLDsaTestData))]
+        public static void ImportExportVerify(CompositeMLDsaTestData.CompositeMLDsaTestVector vector)
+        {
+            using (CompositeMLDsa key = CompositeMLDsa.ImportCompositeMLDsaPrivateKey(vector.Algorithm, vector.SecretKey))
+            {
+                byte[] exportedSecretKey = key.ExportCompositeMLDsaPrivateKey();
+                // TODO 'D' values differ, so we cannot compare keys directly
```

Comment:
Yep, that's a bit of a jerk.

```C#
if (!exported.SequenceEqual(expected))
{
    byte[] exported2 = export again();
    AssertExtensions.SequenceEqual(exported, exported2);

    using (imported2 = import(exported))
    {
        byte[] alternateSig = imported2.Sign(data);
        AssertExtensions.TrueExpression(key.Verify(data, alternateSig));
    }
}
```

is about the best I can think of.

Alternatively, if they're not sequence equal, and the family is RSA, make the key-equality thing in the RSA tests a static method, and call it.